src/material/slide-toggle/slide-toggle.scss

```diff
@@ -131,8 +131,20 @@ $mat-slide-toggle-bar-track-width: $mat-slide-toggle-bar-width - $mat-slide-togg
   // Prevent shrinking of the bar container. It can happen that the content is long enough to
   // shrink the bar and the thumb.
   flex-shrink: 0;
-
   border-radius: $mat-slide-toggle-bar-border-radius;
+
+  // Used to provide the background color for the toggle bar. We need a separate (pseudo) element
+  // for this, because we need to fall back to using `rgb` and `opacity` for the background if
+  // the theme has some values that are provided as CSS variables.
+  &::before {
+    content: '';
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    border-radius: inherit;
+  }
```

Comment:
Can't we use the actual `mat-slide-toggle-bar` element and apply the opacity there if needed? I might miss something, but what is special about the pseudo element, that we cannot do on the actual bar element?

Reply:
That's what I did initially, but the slide toggle thumb is inside the bar so it'll also get the reduced opacity.

Reply:
Very good point. Can you please add that to the comment? It was not obvious to me.

Reply:
Done.